tests/test_options.py

```diff
@@ -1455,31 +1688,252 @@ def cli(one, two):
         runner.invoke(cli, [])
 
 
+def test_custom_type_click_class_flag_value(runner):
+    """A reproduction of
```

Comment:
Describe what the test is doing in the docstring, the link should be extra context only.

Reply:
Done in: https://github.com/pallets/click/pull/3030/commits/c498ced0e7b595880b70a81767ae86902714ee1c